networkx/algorithms/similarity.py

```diff
@@ -1270,7 +1270,11 @@ def avg_sim(s):
         return sum(newsim[w][x] for (w, x) in s) / len(s) if s else 0.0
 
     def sim(u, v):
-        return importance_factor * avg_sim(list(product(G[u], G[v])))
+        if nx.classes.function.is_directed(G):
```

Comment:
Maybe combine avg_sim and sim using something like:

```python
Gadj = G.pred if G.is_directed() else G.adj
def sim(u, v):
    pairs = list(product(Gadj[u], Gadj[v]))
    if not pairs:
        return 0.0
    return importance_factor * sum(newsim[w][x] for w, x in pairs) / len(pairs)
```

Reply:
@dschult Cool to learn about `G.pred` and `G.adj`.  I change the conditional per your recommendation and pushed the change.

At first I also combined the functions, but as a new user of NetworkX my opinion is this made the SimRank function harder to read and understand.  I left them as separate for now so you can review.  If it is more in line with NetworkX style to combine the functions, let me know and I will make that change, too.